src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -771,7 +771,19 @@ public int Detach()
         {
             _breakpointManager.ClearBoundBreakpoints();
 
-            _pollThread.RunOperation(() => _debuggedProcess.CmdDetach());
+            try
+            {
+                _pollThread.RunOperation(() => _debuggedProcess.CmdDetach());
+            }
+            catch (DebuggerDisposedException e)
+            {
+                // Detach command could cause DebuggerDisposedException and we ignore that.
+                if (e.AbortedCommand != "-target-detach")
```

Comment:
From the [msdn release article](https://blogs.msdn.microsoft.com/csharpfaq/2014/11/20/new-features-in-c-6/):

> If the parenthesized expression after ‘if’ evaluates to true, the catch block is run, otherwise the exception keeps going. Exception filters are preferable to catching and rethrowing because they leave the stack unharmed. If the exception later causes the stack to be dumped, you can see where it originally came from, rather than just the last place it was rethrown.

It's also more compact and looks nicer in my opinion.